lib/ObjCClientGenerator.php

```diff
@@ -954,7 +954,10 @@ function writePlugin($pluginName, $enumNodes, $classNodes, $serviceNodes, $servi
 		$services = array();
 		foreach($serviceNamesNodes as $serviceNode)
 		{
-			if($serviceNode->hasAttribute('plugin') && $pluginName == '' || !$this->shouldIncludeService(strtolower($serviceNode->getAttribute("name"))) )
+			$serviceNameShouldInclude = $serviceNode->getAttribute("name");
+			if ($pluginName != '')
+				$serviceNameShouldInclude = $pluginName."_".$serviceNameShouldInclude;
+			if($serviceNode->hasAttribute('plugin') && $pluginName == '' || !$this->shouldIncludeService(strtolower($serviceNameShouldInclude)) )
```

Comment:
Instead of building the id using the plugin-name and the service-name you could simply take the id attribute from $serviceNode. It might be even more correct in case of service name of two words because we use camel cased name and lower cased id.